comm/handlers.go

```diff
@@ -14,65 +18,296 @@ import (
+	fmt.Println("Internal Error " + err.(error).Error())
+	fmt.Println(string(debug.Stack()))
+
+	response.Write([]byte("Server Error " + err.(error).Error()))
 }
 
 // Return an account that matches, or issue an error
 func ValidateRequest(response http.ResponseWriter, request *http.Request) (*accounts.Account, error) {
-	fmt.Println("Validating Request")
 
-	username := accounts.Username() // TODO: Opps!
+	tokenString := request.Header.Get("Authorization")
+	if tokenString == "" {
+		buffer := []byte("Missing Token, Authenication Failed")
+		response.Write(buffer)
+		return nil, errors.New("Missing Token, Authentication Failed")
+	}
+
+	token := accounts.NewToken(tokenString)
 
-	isValid := true
-	if !isValid {
-		buffer := []byte("Authenication Failed")
+	if !accounts.Validate(token) {
+		buffer := []byte("Invalid Token, Authenication Failed")
 		response.Write(buffer)
-		return nil, errors.New("Authentication Failed")
+		return nil, errors.New("Invalid Token, Authentication Failed")
 	}
 
-	root := server()
-	account := root.Auth.Find(username)
-	if account == nil {
+	username := request.Header["Name"][0]
+	if username == "" {
 		fmt.Println("Missing Account", username)
-		log.Dump("Authentication", root.Auth)
+
 		return nil, errors.New("User not Registered")
 	}
 
-	fmt.Println("Validated Account Access for ", account.Username)
+	account := Global().Auth.Find(username)
+
+	if account == nil {
+		fmt.Println("Missing Account Information ", username)
+
+		return nil, errors.New("User not Registered")
+	}
 
 	return account, nil
 }
 
 // Get the Request structure from HTTP request
-func GetClientRequest(request *http.Request) *Request {
+func GetClientRequest(request *http.Request) *jobs.Request {
 
 	buffer, err := ioutil.ReadAll(request.Body)
 	if err != nil {
 		log.Fatal("Request Body", err)
 	}
 
-	log.Dump("Request Buffer", buffer)
-
-	var prototype Request
+	var prototype jobs.Request
 	result := datastore.Deserialize(buffer, &prototype)
 
-	return result.(*Request)
+	return result.(*jobs.Request)
+}
+
+func StreamResponse(job *jobs.ActiveJob, response http.ResponseWriter) {
+	if job == nil {
+		WriteResponse(response, "Inactive Job")
+		return
+	}
+
+	// Stream what is there now
+	// Watch and stream the rest
+	var err error
+	if job.File == nil {
+		fmt.Println("Opening File", job.Cmd.Filepath)
+
+		job.File, err = os.Open(job.Cmd.Filepath)
+		if err != nil {
+			HandleError(response, err)
+			return
+		}
+
+		// TODO: Pick a good offset if the incomming option was tail
+		job.Offset = 0
+	}
+
+	// Move the start to a later location in the file.
+	if job.Offset != 0 {
+		fmt.Println("Moving Forward")
+		job.Offset, err = job.File.Seek(job.Offset, 0)
+		if err != nil {
+			HandleError(response, err)
+		}
+	}
+
+	// Keep trying until the job stops, it runs out of file, or the pipe closes.
+	finished := false
+	for {
+		time.Sleep(1000 * time.Millisecond)
+		buffer := make([]byte, 1024)
+
+		count, err := job.File.Read(buffer)
```

Comment:
I don't see anywhere that read deadlines are set, so I believe this Read call should be blocking on data being available. It's not necessary to throttle the reads below with sleep calls, since this won't busy loop.